docs: updated and re-structured pull_request_template (#86)
Learn-Build-Teach/deals-for-devs: .github/pull_request_template.md

<!-- Thank you for offering your contribution with Deals for Devs! -->

## Description

> <!-- Briefly explain what this PRs purpose is. -->

Reviewer: [suggestion] It'd be best to describe them as proposed changes. Some PRs get rejected.  
```suggestion
<!-- Briefly describe your PR‘s proposed changes. -->
```

## Issue

> <!-- If this PR is related to a specific issue # please list that here. (e.g - "Closes: XXXX") -->

Author: Updated to match your comments!

- Completed

> ## Aditional Information

Author: Out of curiosity, wouldn't Additional Information allow for more open ended communication instead of only allowing screenshots in that area? Sometimes items may need clarification or points to be addressed? I'm fine with either, I just wanted to voice my opinion before making the change. :)

> ## Aditional Information

Reviewer: Good question @BryanF1nes. In my experience with other projects, having a separate area for screenshots is best because for PRs where visual changes are crucial, maintainers and other contributors can have an easier time reviewing them.

> ## Aditional Information

Author: Good to know! Thanks I'll go ahead and make those changes! Thanks

> <!-- This can be any comments you feel are necessary to share or even images that may be useful. -->

Reviewer: ```suggestion
<!-- Add a screenshot of your contribution here. -->
```

Author: Updated

- Completed

## PR Requirements

> <!-- Within the [ ] remove the space with an x -->

Reviewer: ```suggestion
<!-- Add an x in the [ ] if you have done each task-->
```

Author: Updated

- Completed

- [ ] I have carefully read through and understand the [Deals-for-Devs Contributing Guide](https://github.com/Learn-Build-Teach/deals-for-devs/blob/dev/.github/contributing.md)
- [ ] The title of this PR follows the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) format
- [ ] The `Description` gives a good representation of the changes made
- [ ] If this PR addresses an open Issue, it is linked in the `Issue` section

> <!-- Once again we thank you for your time and effort in helping us alongside this open source project! -->

Reviewer: [suggestion] This section is where contributors add their thoughts, so it'd be best to put this information instead of a thank you note. 
```suggestion
<!-- If you have any additional thoughts, just add them here. -->
```

Author: Updated

- Completed
